pkg/controller/core/workload_controller.go

```diff
@@ -127,6 +127,21 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling Workload")
 
+	if eviction := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); eviction != nil && eviction.Status == metav1.ConditionTrue &&
+		eviction.Reason == kueue.WorkloadEvictedByPodsReadyTimeout {
+		if apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadAdmitted) {
+			log.V(2).Info("Cancelling admission of the workload due to exceeding the PodsReady timeout")
+			err := workload.UnsetAdmissionWithCondition(ctx, r.client, &wl, "Evicted", fmt.Sprintf("Exceeded the PodsReady timeout %s", req.NamespacedName.String()))
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		} else {
+			// finish up the eviction
+			log.V(2).Info("Reset eviction condition")
+			err := workload.UpdateStatus(ctx, r.client, &wl, kueue.WorkloadEvicted, metav1.ConditionFalse,
```

Comment:
We don't need to reset this at this point: the workload is still "evicted".
It should be reset when the Workload is Admitted again, possibly in the same API call (to minimize load on API server)